source/common/http/http1/codec_impl.cc

```diff
@@ -687,7 +692,9 @@ StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder)
 
 void ClientConnectionImpl::onEncodeComplete() {
   // Transfer head request state into the pending response before we reuse the encoder.
-  pending_responses_.back().head_request_ = request_encoder_->headRequest();
+  if (!pending_responses_.empty()) {
```

Comment:
Why is this change needed? Because of the other change? I'm hoping by doing what I suggest in the other change we can avoid this?

Reply:
Since `RequestStreamEncoderImpl` is called in ws_handler_impl, it internally calls`StreamEncoderImpl::encodeHeaders`

```
Http1::RequestStreamEncoderImpl upstream_request = Http1::RequestStreamEncoderImpl(upstream_http);
```
This` RequestStreamEncoderImpl` is not created by `ClientConnectionImpl::newStream`. At this time, the peningding_response inside ClientConnectionImpl does not have any elements, so calling `pending_responses_.back().head_request_ = request_encoder_->headRequest();` will cause problems.

```
StreamEncoder& ClientConnectionImpl::newStream(StreamDecoder& response_decoder) {
  if (resetStreamCalled()) {
    throw CodecClientException("cannot create new streams after calling reset");
  }
  // Streams are responsible for unwinding any outstanding readDisable(true)
  // calls done on the underlying connection as they are destroyed. As this is
  // the only place a HTTP/1 stream is destroyed where the Network::Connection is
  // reused, unwind any outstanding readDisable() calls here.
  while (!connection_.readEnabled()) {
    connection_.readDisable(false);
  }
  request_encoder_.reset(new RequestStreamEncoderImpl(*this));
  pending_responses_.emplace_back(&response_decoder);
  return *request_encoder_;
}
```